framework/src/ics/InitialConditionWarehouse.C

```diff
@@ -33,40 +33,45 @@ InitialConditionWarehouse::initialSetup(THREAD_ID tid)
 void
 InitialConditionWarehouse::addObject(MooseSharedPointer<InitialCondition> object, THREAD_ID tid)
 {
-  // Check that when object is boundary restricted that the variable is nodal
-  const MooseVariable & var = object->variable();
 
-  // Boundary Restricted
-  if (object->boundaryRestricted())
+  // Perform IC error checking, there is no need to do this on each thread, since they are all the same
```

Comment:
I know why, the error checking relies on lists of object names. So, when those lists are checked from multiple threads there are race conditions.